src/Umbraco.Core/Persistence/Repositories/Implement/DocumentVersionRepository.cs

```diff
@@ -55,6 +62,47 @@ public IReadOnlyCollection<ContentVersionCleanupPolicySettings> GetCleanupPolici
             return _scopeAccessor.AmbientScope.Database.Fetch<ContentVersionCleanupPolicySettings>(query);
         }
 
+        /// <inheritdoc />
+        public IEnumerable<ContentVersionMeta> GetPagedItemsByContentId(int contentId, long pageIndex, int pageSize, out long totalRecords, int? languageId = null)
+        {
+            var query = _scopeAccessor.AmbientScope.SqlContext.Sql();
+
+            query.Select(@"umbracoDocument.nodeId as contentId,
+                           umbracoContent.contentTypeId as contentTypeId,
+                           umbracoContentVersion.id as versionId,
+                           umbracoContentVersion.userId as userId,
+                           umbracoContentVersion.versionDate as versionDate,
+                           umbracoDocumentVersion.published as currentPublishedVersion,
+                           umbracoContentVersion.[current] as currentDraftVersion,
+                           umbracoContentVersion.preventCleanup as preventCleanup,
+                           umbracoUser.userName as username")
+                .From<DocumentDto>()
+                .InnerJoin<ContentDto>()
+                    .On<DocumentDto, ContentDto>(left => left.NodeId, right => right.NodeId)
+                .InnerJoin<ContentVersionDto>()
+                    .On<ContentDto, ContentVersionDto>(left => left.NodeId, right => right.NodeId)
+                .InnerJoin<DocumentVersionDto>()
+                    .On<ContentVersionDto, DocumentVersionDto>(left => left.Id, right => right.Id)
+                .LeftJoin<UserDto>()
+                    .On<UserDto, ContentVersionDto>(left => left.Id, right => right.UserId)
+                .LeftJoin<ContentVersionCultureVariationDto>()
+                    .On<ContentVersionCultureVariationDto, ContentVersionDto>(left => left.VersionId, right => right.Id)
+                .Where<ContentVersionDto>(x => x.NodeId == contentId);
+
+            // TODO: If there's not a better way to write this then we need a better way to write this.
+            query = languageId.HasValue
+                ? query.Where<ContentVersionCultureVariationDto>(x => x.LanguageId == languageId.Value)
+                : query.Where("umbracoContentVersionCultureVariation.languageId is null");
```

Comment:
@bergmania maybe I am missing something, but shouldn't the second line use `Cms.Core.Constants.DatabaseSchema.Tables.ContentVersionCultureVariation`?
https://github.com/umbraco/Umbraco-CMS/blob/v9/contrib/src/Umbraco.Core/Persistence/Constants-DatabaseSchema.cs#L28

`query.Where<ContentVersionCultureVariationDto>(x => x.LanguageId  == null);`

or 

`query.Where($"{Cms.Core.Constants.DatabaseSchema.Tables.ContentVersionCultureVariation}.languageId is null");`

Reply:
Ahh, I missed the `umbraco` table prefix.
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Persistence/Constants-DatabaseSchema.cs#L28

Anyway I think it should use either the dto class or the constant.
https://github.com/umbraco/Umbraco-CMS/blob/v9/contrib/src/Umbraco.Core/Persistence/Constants-DatabaseSchema.cs#L28

Reply:
Can't do `query.Where<ContentVersionCultureVariationDto>(x => x.LanguageId == null);` as LanguageId is int.
Using constants sounds smnart.

We should probably just create a sql view and create a poco to represent that -- however sql ce

Reply:
Ahh okay, the constants seems a bit nicer then 😁